Shock/Classes/Middleware/Middleware.swift

```diff
@@ -26,6 +28,7 @@ public protocol MiddlewareResponseContext: class {
 public protocol MiddlewareContext {
     var requestContext: MiddlewareRequestContext { get }
     var responseContext: MiddlewareResponseContext { get }
+    var notFoundHandler: HandlerClosure? { get }
```

Comment:
I'm not sure I like this as it circumvents the standard middleware pattern and is letting NIO implementation details bleed into the context interface  - the middleware shouldn't need to know about special HTTP handling cases; the purpose of middleware is to be generic enough so as to allow the implementor to cover these cases themself.

No need to hold up the PR for this, but I'll have a think about how the same functionality can be achieved without the interface bleed.